docs/guides/references/migration-guide.mdx

```diff
@@ -2,6 +2,68 @@
 title: Migration Guide
 ---
 
+## Migrating to Cypress 13.0
+
+This guide details the changes and how to change your code to migrate to Cypress
+version 13.0.
+[See the full changelog for version 12.0](/guides/references/changelog#12-0-0).
+
+### `cy.readFile()` is now a query command
+
+In Cypress 13, the [`.readFile()`](/api/commands/as) command is now a query.
+Tests written using it should continue to operate exactly as before; no changes
+are necessary.
+
+This means that `readFile()` will re-read the file from disk if any upcoming
+command in the same chain fails. Assertions no longer have to be directly
+attached.
+
+```js
+cy.readFile(`users.json`).its('users.123.fullName').should('eq', 'John Doe')
+```
+
+Beginning with Cypress 13, the above test will re-read the file until the file
+exists, it has the requested property, and it passes the assertion.
+
+In previous versions of Cypress, the above command would retry until the file
+existed, but would _not_ re-read it from disk if the file didn't have the
+requested property or the contents didn't match.
+
+#### `.readFile()` can no longer be overwritten with `Cypress.Commands.overwrite()`
```

Comment:
My hope is to have https://github.com/cypress-io/cypress/issues/25078 done before we release Cy13, which would necessitate rewriting this part of the migration guide (to say "actually you can totally still do it, here's how").